regression/cbmc-incr-smt2/bitvector-bitwise-operators/shift_right.c

```diff
@@ -0,0 +1,24 @@
+int main()
+{
+  // In C, whether a right-shift is arithmetic or logical depends on the
+  // original type being shifted. An unsigned value will be shifted to
+  // the right in a logical manner (this assigns `0` to the leftmost bit).
+  // If the type is signed, right shift will assign the sign bit to the
+  // leftmost digit.
+
+  int bitmask_2 = 0x2;
+  int bitmask_3 = 0x3;
+
+  int first;
+  unsigned int second;
+
+  int place;
+  __CPROVER_assume(place >= 1);
+
+  int result_signed = first >> place;
```

Comment:
We should have tests where `first` is negative and tests where `first` is positive in the output trace with regexes in the `.desc` to confirm that the corresponding values for `result_signed` are as expected. For example if `first` is negative then `result_signed` should be negative.

Reply:
This is resolved.